hi-ml-cpath/src/health_cpath/datasets/base_dataset.py

```diff
@@ -209,8 +214,9 @@ def validate_columns(self) -> None:
         If the constructor is overloaded in a subclass, you can pass `validate_columns=False` and
         call `validate_columns()` after creating derived columns, for example.
         """
-        columns = [self.IMAGE_COLUMN, self.label_column, self.MASK_COLUMN,
-                   self.SPLIT_COLUMN] + list(self.METADATA_COLUMNS)
+        columns = [self.IMAGE_COLUMN, self.label_column, self.MASK_COLUMN, self.SPLIT_COLUMN]
+        if self.validate_metadata:
+            columns.extend(list(self.METADATA_COLUMNS))
```

Comment:
Rather than having an extra flag, could we not just do:
```
if dataframe_kwargs is not empty:
    udpdate`self.METADATA_COLUMNS` to include only the ones that are in dataframe_kwargs
```
at some point in the pipeline?

Reply:
but dataframe_kwargs is not only specific to usecols, it's meant to be used to override all [pd.read_csv](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html) arguments

Reply:
I can maybe check if usecols is in dataframe_kwargs there and maybe validate only the columns defined there?

Reply:
Sorry yes that's what I mean